modules/indigo/module/inc/indigo/of_state_manager.h

```diff
@@ -329,6 +407,24 @@ indigo_core_gentable_register(
 void
 indigo_core_gentable_unregister(indigo_core_gentable_t *gentable);
 
+/*
+ * @brief resume indigo addi4()/modifyi4()/del4() when the driver gets the final status 
+ * @param op_ctx operation context 
+ * @param entry_priv driver's private data for this entry
+ * @param err_txt error string from driver
+ * @param rv the final status of the driver async add/modify
+ *
+ * The rv status will decide how the indigo to do with the rest works.
```

Comment:
"the state manager will use the value of rv to either complete the async operation in case of success or clean up in case of failure."  
actually we should enumerate all cases that the resume function is expected to handle (and also test them in the unit tests).

Reply:
My unit tests covers success and timeout. Actually, the state manager handles the timeout as the other errors.